etl_entities/hwm/hwm.py

```diff
@@ -0,0 +1,148 @@
+#  Copyright 2023 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from datetime import datetime
+from typing import Any, Generic, Optional, TypeVar
+
+from pydantic import Field, validate_model
+
+from etl_entities.entity import GenericModel
+from etl_entities.hwm_utils import HWMTypeRegistry
+
+ValueType = TypeVar("ValueType")
+
+
+class HWM(ABC, Generic[ValueType], GenericModel):
+    """Generic HWM type
+
+    Parameters
+    ----------
+    column : ``str``
+
+        Column name
+
+    name : ``str``
+
+        Table name
+
+    value : ``ColumnValueType`` or ``None``, default: ``None``
+
+        HWM value
+
+    description: ``str``, default: ``""``
+
+        Description of HWM
+
+    modified_time : :obj:`datetime.datetime`, default: current datetime
+
+        HWM value modification time
+    """
+
+    name: str
+    value: Optional[ValueType]
```

Comment:
```suggestion
    value: ValueType
```

Не у всех типов HWM значение может быть None. У FileListHWM список файлов, при создании нового объекта это пустой список, а не None.